### ISSUE TITLE
Importing nodes?

### ISSUE THREAD

**Issue by festro** [leon-thomm/Ryven#16]
I know this isn't a bug but couldn't find anywhere else to ask.
as for the question, I was wondering if it was possible to import nodes from modules without having to define or create them from scratch?

**Comment by leon-thomm** [comment]
Hi,
do you mean automatically creating nodes from python modules (like _random_)? If so, it's not quite the main idea in Ryven. If you take a look at the existing nodes, you will see that most of them do much more than representing one single command. The idea isn't to replace textual coding but to enhance it by the visual representation in a level of abstraction that focuses of what's important for the script. For example, instead of writing a huge visual script to fetch some emails for further processing, you may want to put all that into very few nodes. Also, you could not automate the process of creating intuitive GUI.
There certainly are a lot of python modules with functionality that, at least as a start, would be useful to be accessible through automatically generated nodes. But adding some kind of dynamics would really make things complicated. And if it's just about simple function calls from modules (like _random()_), creating nodes for them takes seconds once you got some routine.
Does that address your question?

**Comment by festro** [comment]
Was thinking of something more like import so functions from a module like NumPy could automatically be converted into nodes without having to reverse engineer the code to make it from scratch (which would be impossible for someone of my skill) in the node creator.

**Comment by leon-thomm** [comment]
Isn't this exactly what I just suggested? Or am I barking up the wrong tree? As can be read on the website: _Ryven is not a tool to learn how to program or to generally program in Python_. I don't suggest using Ryven to create visual scripts if you don't know what they do. That would be the approach to completely hide the source code level and replace it with the visual representation, which is not the idea in Ryven. I've found, that this only works if you create a whole ecosystem of software supporting your process of scripting (basically an IDE) which is far beyond the bounds of possibility for me, so I've set a different focus. Blueprints in UnrealEngine are an excellent example of the opposite approach, but they actually did create a whole IDE for programming visually there.
